tools/ports/__init__.py

```diff
@@ -71,6 +88,30 @@ def init_port(name, port):
   validate_port(port)
 
 
+def load_external_port(external_port):
+  need_clear_project_build = Ports.fetch_project(external_port.name, external_port.EXTERNAL_PORT, external_port.SHA512, True)
+  port_file = os.path.join(Ports.get_dir(), external_port.name, external_port.PORT_FILE)
+  module_name = f'tools.ports.external.{external_port.name}'
+  spec = importlib.util.spec_from_file_location(module_name, port_file)
+  local_port = importlib.util.module_from_spec(spec)
+  spec.loader.exec_module(local_port)
+  ports.remove(external_port)
+  for a in ['URL', 'DESCRIPTION', 'LICENSE']:
+    if not hasattr(local_port, a):
+      setattr(local_port, a, getattr(external_port, a))
+  init_port(external_port.name, local_port)
+  if need_clear_project_build:
+    Ports.clear_project_build(external_port.name)
```

Comment:
Why does this need to be delayed like this?

Reply:
The problem is that this function (`load_external_port`) calls `Ports.fetch_project` which then can calls`Ports.clear_project_build` which then calls `get_port_by_name` which then calls `load_external_port` again.

Meaning it is a recursive call... 

I ran into this issue after clearing the cache manually. By delaying the call until the end, it fixes this issue.